develop/dev-guide-sample-application-nodejs-mysql2.md

```diff
@@ -0,0 +1,311 @@
+         minVersion: 'TLSv1.2',
+         ca: process.env.TIDB_CA_PATH ? fs.readFileSync(process.env.TIDB_CA_PATH) : undefined
+      } : null,
+   }
+   const conn = await createConnection(options);
+
+   // Step 4. Perform some SQL operations...
+
+   // Step 5. Close the connection.
+   await conn.end();
+}
+
+void main();
+```
+
+> **Note**
+>
+> For TiDB Serverless, TLS connection **MUST** be enabled via `TIDB_ENABLE_SSL` when using public endpoint, but you **don't** have to specify an SSL CA certificate via `TIDB_CA_PATH`, because Node.js uses the built-in [Mozilla CA certificate](https://wiki.mozilla.org/CA/Included_Certificates) by default, which is trusted by TiDB Serverless.
+
+### Insert data
+
+The following query creates a single `Player` record and returns a `ResultSetHeader` object:
+
+```javascript
+const [rsh] = await conn.query('INSERT INTO players (coins, goods) VALUES (?, ?);', [100, 100]);
+console.log(rsh.insertId);
+```
+
+For more information, refer to [Insert data](/develop/dev-guide-insert-data.md).
+
+### Query data
+
+The following query returns a single `Player` record by ID `1`:
+
+```javascript
+const [rows] = await conn.query('SELECT id, coins, goods FROM players WHERE id = ?;', [1]);
+console.log(rows[0]);
+```
+
+For more information, refer to [Query data](/develop/dev-guide-get-data-from-single-table.md).
+
+### Update data
+
+The following query adds `50` coins and `50` goods to the `Player` with ID `1`:
+
+```javascript
+const [rsh] = await conn.query(
+    'UPDATE players SET coins = coins + ?, goods = goods + ? WHERE id = ?;',
+    [50, 50, 1]
+);
+console.log(rsh.affectedRows);
+```
+
+For more information, refer to [Update data](/develop/dev-guide-update-data.md).
+
+### Delete data
+
+The following query deletes the `Player` record with ID `1`:
+
+```javascript
+const [rsh] = await conn.query('DELETE FROM players WHERE id = ?;', [1]);
+console.log(rsh.affectedRows);
+```
+
+For more information, refer to [Delete data](/develop/dev-guide-delete-data.md).
+
+## Useful notes
+
+- Using [connection pools](https://github.com/sidorares/node-mysql2#using-connection-pools) to manage database connections can reduce the performance overhead caused by frequently establishing and destroying connections.
+- To avoid SQL injection, it is recommended to use [prepared statements](https://github.com/sidorares/node-mysql2#using-prepared-statements).
+- In scenarios where there are not many complex SQL statements involved, using ORM frameworks like [Sequelize](https://sequelize.org/), [TypeORM](https://typeorm.io/), or [Prisma](https://www.prisma.io/) can greatly improve development efficiency.
+- Enable the `supportBigNumbers: true` option when dealing with big numbers (`BIGINT` and `DECIMAL` columns) in the database.
+- Enable the `enableKeepAlive: true` option to avoid socket error `read ECONNRESET` due to network problems. (Related issue: [sidorares/node-mysql2#683](https://github.com/sidorares/node-mysql2/issues/683))
```

Comment:
```suggestion
- It is recommended to enable the `enableKeepAlive: true` option to avoid socket error `read ECONNRESET` due to network problems. (Related issue: [sidorares/node-mysql2#683](https://github.com/sidorares/node-mysql2/issues/683))
```